src/network/url_downloader.cpp

```diff
@@ -82,7 +82,12 @@ download(QNetworkAccessManager* manager, const Time& timeout, QUrl const& url, P
     QTimer download_timeout;
     download_timeout.setInterval(timeout);
 
-    QNetworkRequest request{url};
+    QUrl adjusted_url{url};
+
+    if (adjusted_url.scheme() == "http")
+        adjusted_url.setScheme("https");
+
+    QNetworkRequest request{adjusted_url};
```

Comment:
Although you are not adding much, this function is big enough as it is. This is a chance to improve on that a little. Would you mind extracting the request creation into a separate function? Basically lines 85-97. Thanks!